src/wallet/api.rs

```diff
@@ -1267,193 +1270,237 @@ impl Wallet {
+            match select_coin(&single_output_groups, &coin_selection_option) {
+                Ok(selection) => {
+                    let additional_utxos: Vec<_> = selection
+                        .selected_inputs
+                        .iter()
+                        .flat_map(|&group_index| single_addresses[group_index].clone())
+                        .collect();
 
-        // Run coin selection on single addresses only
-        match select_coin(&single_output_groups, &coin_selection_option) {
-            Ok(selection) => {
-                let additional_utxos: Vec<_> = selection
-                    .selected_inputs
-                    .iter()
-                    .flat_map(|&group_index| single_addresses[group_index].clone())
-                    .collect();
-
-                // Combine pre-selected groups + coin selection results
-                let mut final_selection = selected_utxos;
-                final_selection.extend(additional_utxos);
-
-                log::info!("Coin selection concluded with {:?}", selection.waste);
-                Ok(final_selection)
+                    // Combine pre-selected groups + coin selection results
+                    let mut final_selection = selected_utxos;
+                    final_selection.extend(additional_utxos);
+
+                    log::info!("Selected {} {} UTXOs", final_selection.len(), utxo_type);
+                    return Ok(final_selection);
+                }
+                Err(_) => {
+                    // Continue to next UTXO type
+                    continue;
+                }
             }
-            Err(e) if e.to_string().contains("The Inputs funds are insufficient") => {
-                let total_available = selected_total
-                    + single_addresses
-                        .iter()
-                        .flatten()
-                        .map(|(u, _)| u.amount.to_sat())
-                        .sum::<u64>();
+        }
+
+        // Both failed
+        if target_sats > regular_total && target_sats > swap_total {
+            let total_available = regular_total + swap_total;
+            if total_available < target_sats {
                 Err(WalletError::InsufficientFund {
                     available: total_available,
-                    required: amount.to_sat(),
+                    required: target_sats,
                 })
+            } else {
+                Err(WalletError::General(format!(
+                    "Coin selection failed with separated UTXO types: regular_total={} sats, swap_total={} sats, target={} sats. Neither type alone can satisfy the target.",
+                    regular_total, swap_total, target_sats
+                )))
             }
-            Err(e) => Err(WalletError::General(format!("Coin selection failed: {e}"))),
+        } else {
+            Err(WalletError::General(
+                "Coin selection failed for unknown reason".to_string(),
+            ))
```

Comment:
The total available concept is redundant now.  We should just say here that Coinselection failed, mentioning, target, regualr_total and swap_total values. 

Also we can simply add this check at the very start of the CS, to check against the target with regular and swap total. If that fails theres no point of doing anything and we can error early.

Then the loop should be the last thing in the function. No error handling required after the loop.